dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/ds3treetable/Ds3TreeTablePresenter.java

```diff
@@ -498,23 +542,55 @@ public void changed(final ObservableValue observable, final Object oldValue,
                         ds3TreeTable.getSelectionModel().select(row.getIndex());
                     } else {
                         ds3TreeTable.getSelectionModel().setSelectionMode(SelectionMode.MULTIPLE);
+                        final int selectedSize = ds3TreeTable.getSelectionModel().getSelectedItems().size();
                         ds3TreeTable.getSelectionModel().clearSelection(row.getIndex());
                         rowNameList.remove(row.getTreeItem().getValue().getName());
+                        manageItemsCount(row.getTreeItem());
                     }
                 } else if (event.getClickCount() == 2) {
+                    final ProgressIndicator progress = new ProgressIndicator();
```

Comment:
This method is starting to have too much complexity.  Please split up into smaller components.